tests/unit/src/test/scala/tests/MtagsSuite.scala

```diff
@@ -13,35 +14,34 @@ import scala.meta.internal.semanticdb.Scala._
  * It turns out ScalaMtags is actually more correct semanticdb-scalac when
  * it comes to trickier cases like implicit conversions and pattern matching.
  */
-class MtagsSuite extends DirectoryExpectSuite("mtags") {
+abstract class MtagsSuite(
+    inputProperties: => InputProperties,
+    directory: String,
+    dialect: Dialect,
+    exclude: InputFile => Boolean,
+    ignoreUnknwonSymbols: Boolean = false,
+    ignoreAssertion: Boolean = false
+) extends DirectoryExpectSuite(directory) {
 
-  def hasSemanticdbBug(file: InputFile): Boolean = {
-    // don't assert fidelity where semanticdb-scalac has known bugs and mtags is correct.
-    List(
-      "ImplicitClasses",
-      "PatternMatching",
-      "ImplicitConversions",
-      "MacroAnnotation"
-    ).exists { name => file.file.toNIO.endsWith(s"$name.scala") }
-  }
+  override lazy val input: InputProperties = inputProperties
 
   def testCases(): List[ExpectTestCase] = {
     input.allFiles.map { file =>
       ExpectTestCase(
         file,
         { () =>
           val input = file.input
-          val mtags = Mtags.index(input, dialects.Scala213)
+          val mtags = Mtags.index(input, dialect)
           val obtained = Semanticdbs.printTextDocument(mtags)
           val unknownSymbols = mtags.occurrences.collect {
             case occ if symtab.info(occ.symbol).isEmpty =>
               val pos = input.toPosition(occ)
               pos.formatMessage("error", s"unknown symbol: ${occ.symbol}")
           }
-          if (unknownSymbols.nonEmpty) {
+          if (!ignoreUnknwonSymbols && unknownSymbols.nonEmpty) {
             fail(unknownSymbols.mkString("\n"))
           }
-          if (file.isScala && !hasSemanticdbBug(file)) {
+          if (file.isScala && !ignoreAssertion && !exclude(file)) {
```

Comment:
`ignoreAssertion` leads to that MtagsScala3Suite doesn't perform any check.

Reply:
Yes, if `ignoreAssertion = true` MtagsSuite doesn't do any checks, just generates a result into the designated directory.

Reply:
> I mean, if there are no checks what is the purpose of this test?
> I think it would be better tot turn it into an issue and introduce MtagsScala3Suite later.


I will turn it into an issue, and make `MtagsScala3Suite` actually works later (in another PR).
Indeed, `MtagsScala3Suite` doesn't have any checks, and looks like useless, I believe generating expected files will be helpful for us to check how mtags (for Scala3) generate `SymbolOccurrence`s.
With these generated files, we can see how they're different from scalac-semanticdb generated semanticdb, and tell what's the issue.

Reply:
Ok, now I see the point. Here we declare how it works now it be able to spot issue if we break ScalaMtags  in future but skip the one-to-one comparison with semanticdb.

So, then maybe simplify it a bit.
Instead of introducing `ignoreAssertion` parameter we can use `exclude`:
```scala
class MtagsScala3Suite
    extends MtagsSuite(
      InputProperties.scala3(),
      "mtags-scala3",
      dialects.Scala3,
      _ => true, // exclude all
      // Do not assert unknown symbols and compare to semanticdb,
      // There're still lot to improve in mtags for Scala3
      ignoreUnknownSymbols = true,
    )
```
I would also rename `exclude` to some other name (`compareWithSemanticdb` ?) to express that it isn't fully excluded but just doesn't equal to semanticdb.